src/microsim/schema/simulation.py

```diff
@@ -168,6 +186,104 @@ def _truth_cache_path(
             truth_cache = truth_cache / f"seed{seed}"
         return truth_cache
 
+    def get_wavelength_bins(self) -> list[Bin]:  # TODO: private?
+        """
+        Create wavelength bins depending on flurophores emission spectra.
+
+        NOTE: we assume this is independent of the excitation/emission filters
+        applied, so we only consider the emission spectra of the fluorophores.
+        Why? Because the excitation filters are applied to the light source, and
+        emission filters are after illumination of fluorophores, for which we need
+        matching bins.
+        """
+        fluorophores = [x.fluorophore for x in self.sample.labels]
+        if len(fluorophores) == 0:
+            # we have no fluorophores to calculate
+            raise ValueError("No fluorophores in the current sample!")
+
+        # Get emission spectra for all the fluorophores
+        fluor_em_spectra = []
+        for fluor in fluorophores:
+            if fluor is None:
+                raise NotImplementedError(
```

Comment:
this is going to fail tests at the moment.  so before this can merge we need to come up with some heuristic.   This is a bit tricky of course, and I don't expect you to have to come up with a "proper" solution here.  however, someone *should* be able to run a simulation with a bare number of parameters like this:

```python
sim = ms.Simulation(
    truth_space=ms.ShapeScaleSpace(shape=(128, 512, 512), scale=(0.02, 0.01, 0.01)),
    output_space={'downscale': 8},
    sample=ms.Sample(
        labels=[ms.MatsLines(density=0.5, length=30, azimuth=5, max_r=1)]
    ),
    modality=ms.Confocal(pinhole_au=0.2),
)
```

and only later start caring about the specifics of the fluorophore.

perhaps implement a hack like we do in `emission_flux` where `if fluor is None`.  just to get something running

Reply:
I thought about the following:

- In `Simulation.settings`, we add 2 more attributes: `min_wavelength` and `max_wavelength`. These parameters are used to perform binning in case fluorophores are not specified. For instance, we could also use these [here](https://github.com/tlambert03/microsim/blob/main/src/microsim/schema/simulation.py#L194) instead of hardcoding some lower and upper bounds.
- Then, [here](https://github.com/tlambert03/microsim/pull/63/files#diff-28718170e2fde3b2fdd052311e35a5101e9989d64fc99d466e0b0a0ba6d7ee2cR208) we append `np.array([min_wave, max_wave])` to `fluor_em_spectra`.

Reply:
If you think this is fine, I will push this new version too :)

Reply:
See: [here](https://github.com/tlambert03/microsim/pull/63/files#diff-6ff3c232597b244078c24c33ea56f90c8fb04c6ecef702d51b2967a20cddb773R65-R80) for definition of new attributes, [here](https://github.com/tlambert03/microsim/pull/63/files#diff-28718170e2fde3b2fdd052311e35a5101e9989d64fc99d466e0b0a0ba6d7ee2cR252-R261) for handling of no-FP situation in `illumination_flux`, and [here](https://github.com/tlambert03/microsim/pull/63/files#diff-28718170e2fde3b2fdd052311e35a5101e9989d64fc99d466e0b0a0ba6d7ee2cR303-R308) for changes in `emission_flux`.